src/Lykke.Service.BlockchainHistoryReader.AzureRepositories/Implementations/RepositoryBase.cs

```diff
@@ -0,0 +1,39 @@
+using Lykke.AzureStorage.Tables.Entity.Metamodel;
+using Lykke.AzureStorage.Tables.Entity.Metamodel.Providers;
+
+
+namespace Lykke.Service.BlockchainHistoryReader.AzureRepositories.Implementations
+{
+    public abstract class RepositoryBase
+    {
+        private static readonly object InitLock = new object();
+        
+        private static bool _initialized;
+        
+        
+        
+        protected RepositoryBase()
```

Comment:
You could use static ctor instead of lock object and afaik, AnnotationsBasedMetamodelProvider is configured by default